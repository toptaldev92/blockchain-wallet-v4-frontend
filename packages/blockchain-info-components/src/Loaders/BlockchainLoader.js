import { Image } from '../..'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const Img = styled(Image)`
  position: relative;
  width: ${props => props.width};
  height: ${props => props.height};
  overflow: hidden;
`

const BlockchainLoader = props => <Img {...props} name='logo-loader' />

BlockchainLoader.propTypes = {
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired
}

BlockchainLoader.defaultProps = {
  width: '150px',
  height: '150px'
}

export default BlockchainLoader
