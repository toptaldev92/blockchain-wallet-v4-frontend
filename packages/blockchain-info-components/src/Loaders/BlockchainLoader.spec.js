import { shallow } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

import BlockchainLoader from './BlockchainLoader'

describe('BlockchainLoader component', () => {
  it('default renders correctly', () => {
    const component = shallow(<BlockchainLoader height='10px' width='10px' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
