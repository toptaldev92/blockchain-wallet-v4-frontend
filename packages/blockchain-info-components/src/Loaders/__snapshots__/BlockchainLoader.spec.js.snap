// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockchainLoader component default renders correctly 1`] = `
<styled.div
  height="10px"
  width="10px"
>
  <styled.div>
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
    <Styled(Styled(styled.div)) />
  </styled.div>
</styled.div>
`;
